### ISSUE TITLE
Unable to modify / delete services after composer update

### ISSUE THREAD

**Issue by jackdpeterson** [zfcampus/zf-apigility-admin-ui#115]
Similar error when attempting to delete a service as well.
![screen shot 2016-08-05 at 11 12 34 pm 2](https://cloud.githubusercontent.com/assets/938961/17455087/8c0a067c-5b62-11e6-8d47-ceabfc5adbee.png)

Below, note the duplicated route information (namespace, possibly?).

![screen shot 2016-08-05 at 11 13 07 pm 3](https://cloud.githubusercontent.com/assets/938961/17455096/a4b3c00a-5b62-11e6-867b-77f5ea2455b5.png)

modules.config.php:
```
<?php
return array(
    'Application',
    'ZF\\DevelopmentMode',
    'ZF\\Apigility',
    'ZF\\Apigility\\Provider',
    'ZF\\Apigility\\Documentation',
    'AssetManager',
    'ZF\\ApiProblem',
    'ZF\\Configuration',
    'ZF\\OAuth2',
    'ZF\\MvcAuth',
    'ZF\\Hal',
    'ZF\\ContentNegotiation',
    'ZF\\ContentValidation',
    'ZF\\Rest',
    'ZF\\Rpc',
    'ZF\\Versioning',
    'test',
);
```

Composer:
```
bshaffer/oauth2-server-php          v1.8.0             OAuth2 Server for PHP
container-interop/container-interop 1.1.0              Promoting the interoperability of container objects (DIC, SL, etc.)
doctrine/instantiator               1.0.5              A small, lightweight utility to instantiate objects in PHP without invoking their constructors
herrera-io/json                     1.0.3              A library for simplifying JSON linting and validation.
herrera-io/phar-update              1.0.3              A library for self-updating Phars.
justinrainbow/json-schema           1.6.1              A library to validate a json schema.
kherge/version                      1.0.1              A parsing and comparison library for semantic versioning.
kriswallsmith/assetic               v1.2.1             Asset Management for PHP
michelf/php-markdown                1.6.0              PHP Markdown
myclabs/deep-copy                   1.5.1              Create deep copies (clones) of your objects
phpdocumentor/reflection-common     1.0                Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock   3.1.0              With this component, a library can provide support for annotations via DocBlocks or otherwise retrieve information that is embedded in a DocBl...
phpdocumentor/type-resolver         0.2                
phpspec/prophecy                    v1.6.1             Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage           4.0.1              Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator           1.4.1              FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template           1.2.1              Simple template engine.
phpunit/php-timer                   1.0.8              Utility class for timing
phpunit/php-token-stream            1.4.8              Wrapper around PHP's tokenizer extension.
phpunit/phpunit                     5.5.0              The PHP Unit Testing framework.
phpunit/phpunit-mock-objects        3.2.3              Mock Object library for PHPUnit
psr/http-message                    1.0                Common interface for HTTP messages
psr/log                             1.0.0              Common interface for logging libraries
rwoverdijk/assetmanager             1.6.0              An assetmanager module for Zend Framework 2.
sebastian/code-unit-reverse-lookup  1.0.0              Looks up which function or method a line of code belongs to
sebastian/comparator                1.2.0              Provides the functionality to compare PHP values for equality
sebastian/diff                      1.4.1              Diff implementation
sebastian/environment               1.3.7              Provides functionality to handle HHVM/PHP environments
sebastian/exporter                  1.2.2              Provides the functionality to export PHP variables for visualization
sebastian/global-state              1.1.1              Snapshotting of global state
sebastian/object-enumerator         1.0.0              Traverses array structures and object graphs to enumerate all referenced objects
sebastian/recursion-context         1.0.2              Provides functionality to recursively process PHP variables
sebastian/resource-operations       1.0.0              Provides a list of PHP built-in functions that operate on resources
sebastian/version                   1.0.6              Library that helps with managing the version number of Git-hosted PHP projects
seld/jsonlint                       1.4.0              JSON Linter
symfony/process                     v2.8.9             Symfony Process Component
symfony/yaml                        v3.1.3             Symfony Yaml Component
webmozart/assert                    1.0.2              Assertions to validate method input/output with nice error messages.
zendframework/zend-authentication   2.5.3              provides an API for authentication and includes concrete authentication adapters for common use case scenarios
zendframework/zend-barcode          2.6.0              provides a generic way to generate barcodes
zendframework/zend-cache            2.7.1              provides a generic way to cache any data
zendframework/zend-captcha          2.6.0              
zendframework/zend-code             2.6.3              provides facilities to generate arbitrary code using an object oriented interface
zendframework/zend-config           2.6.0              provides a nested object property based user interface for accessing this configuration data within application code
zendframework/zend-console          2.6.0              
zendframework/zend-crypt            2.6.0              
zendframework/zend-db               2.8.1              
zendframework/zend-debug            2.5.1              
zendframework/zend-developer-tools  dev-master 4d8a2e4 Module for developer and debug tools for use with zend-mvc applications.
zendframework/zend-di               2.6.1              
zendframework/zend-diactoros        1.3.5              PSR HTTP Message implementations
zendframework/zend-dom              2.6.0              provides tools for working with DOM documents and structures
zendframework/zend-escaper          2.5.2              
zendframework/zend-eventmanager     2.6.3              
zendframework/zend-feed             2.7.0              provides functionality for consuming RSS and Atom feeds
zendframework/zend-file             2.7.0              
zendframework/zend-filter           2.7.1              provides a set of commonly needed data filters
zendframework/zend-form             2.9.0              
zendframework/zend-http             2.5.4              provides an easy interface for performing Hyper-Text Transfer Protocol (HTTP) requests
zendframework/zend-hydrator         1.1.0              
zendframework/zend-i18n             2.7.3              
zendframework/zend-i18n-resources   2.5.2              Provides validator translations for zend-i18n's Translator
zendframework/zend-inputfilter      2.7.2              
zendframework/zend-json             2.6.1              provides convenience methods for serializing native PHP to JSON and decoding JSON to native PHP
zendframework/zend-loader           2.5.1              
zendframework/zend-log              2.9.0              component for general purpose logging
zendframework/zend-mail             2.7.1              provides generalized functionality to compose and send both text and MIME-compliant multipart e-mail messages
zendframework/zend-math             2.7.0              
zendframework/zend-memory           2.5.2              
zendframework/zend-mime             2.6.0              
zendframework/zend-modulemanager    2.7.2              
zendframework/zend-mvc              2.7.10             
zendframework/zend-navigation       2.8.1              provides support for managing trees of pointers to web pages
zendframework/zend-paginator        2.7.0              
zendframework/zend-permissions-acl  2.6.0              provides a lightweight and flexible access control list (ACL) implementation for privileges management
zendframework/zend-permissions-rbac 2.5.1              provides a role-based access control management
zendframework/zend-progressbar      2.5.2              component to create and update progressbars in different environments
zendframework/zend-psr7bridge       0.2.2              PSR-7 <-> Zend\Http bridge
zendframework/zend-serializer       2.8.0              provides an adapter based interface to simply generate storable representation of PHP types by different facilities, and recover
zendframework/zend-server           2.7.0              
zendframework/zend-servicemanager   2.7.6              
zendframework/zend-session          2.7.3              manage and preserve session data, a logical complement of cookie data, across multiple page requests by the same client
zendframework/zend-soap             2.6.0              
zendframework/zend-stdlib           2.7.7              
zendframework/zend-tag              2.6.1              a component suite which provides a facility to work with taggable Items
zendframework/zend-test             2.6.1              
zendframework/zend-text             2.6.0              
zendframework/zend-uri              2.5.2              a component that aids in manipulating and validating » Uniform Resource Identifiers (URIs)
zendframework/zend-validator        2.8.1              provides a set of commonly needed validators
zendframework/zend-version          2.5.1              
zendframework/zend-view             2.8.1              provides a system of helpers, output filters, and variable escaping
zendframework/zend-xmlrpc           2.6.0              
zendframework/zenddiagnostics       v1.0.9             A set of components for performing diagnostic tests in PHP applications
zendframework/zendframework         2.5.3              Zend Framework 2
zendframework/zendxml               1.0.2              Utility library for XML usage, best practices, and security in PHP
zendframework/zftool                dev-master 840c3ee Utility module for Zend Framework 2 applications.
zfcampus/zf-api-problem             1.2.1              ZF2 Module providing API-Problem assets and rendering
zfcampus/zf-apigility               1.3.0              Apigility module for Zend Framework
zfcampus/zf-apigility-admin         1.4.3              Apigility Admin module
zfcampus/zf-apigility-admin-ui      1.3.0              Apigility Admin UI module
zfcampus/zf-apigility-documentation 1.2.1              Apigility API documentation module
zfcampus/zf-apigility-provider      1.2.0              Apigility interfaces
zfcampus/zf-configuration           1.2.0              Zend Framework module providing a REST resource for manipulating configuration
zfcampus/zf-console                 1.3.0              Library for creating and dispatching console commands
zfcampus/zf-content-negotiation     1.2.2              ZF2 Module providing content-negotiation features
zfcampus/zf-content-validation      1.3.4              Zend Framework module providing incoming content validation
zfcampus/zf-deploy                  1.2.0              Deployment tool for Zend Framework applications
zfcampus/zf-development-mode        2.1.2              ZF2 development mode module
zfcampus/zf-hal                     1.4.2              ZF2 Module providing Hypermedia Application Language assets and rendering
zfcampus/zf-mvc-auth                1.4.2              ZF2 Module providing Authentication and Authorization events and infrastructure
zfcampus/zf-oauth2                  1.4.0              ZF2 module for implementing an OAuth2 server
zfcampus/zf-rest                    1.3.1              ZF Module providing structure for RESTful resources
zfcampus/zf-rpc                     1.3.0              ZF2 Module for simplifying the creation of RPC services
zfcampus/zf-versioning              1.2.0              ZF2 Module providing listeners and route prototypes for implementing API versioning
```

**Comment by jackdpeterson** [comment]
Steps to reproduce:

1. install apigility
```
curl -sS https://apigility.org/install | php
```

2. create API, Service, modify service, delete service [all work okay, as expected]
3. perform composer.phar update

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Removing rwoverdijk/assetmanager (1.5.0)
  - Removing kriswallsmith/assetic (v1.2.1)
  - Removing symfony/process (v2.8.2)
  - Removing zendframework/zend-stdlib (2.7.4)
  - Installing zendframework/zend-stdlib (2.7.7)
    Loading from cache

  - Removing zendframework/zend-hydrator (1.0.0)
  - Installing zendframework/zend-hydrator (1.1.0)
    Loading from cache

  - Removing zendframework/zend-validator (2.5.3)
  - Installing zendframework/zend-validator (2.8.1)
    Loading from cache

  - Removing zendframework/zend-servicemanager (2.7.4)
  - Installing zendframework/zend-servicemanager (2.7.6)
    Loading from cache

  - Removing zendframework/zend-barcode (2.5.2)
  - Installing zendframework/zend-barcode (2.6.0)
    Loading from cache

  - Removing zendframework/zend-eventmanager (2.6.2)
  - Installing zendframework/zend-eventmanager (2.6.3)
    Loading from cache

  - Removing zendframework/zend-cache (2.5.3)
  - Installing zendframework/zend-cache (2.7.1)
    Loading from cache

  - Removing zendframework/zend-math (2.5.2)
  - Installing zendframework/zend-math (2.7.0)
    Loading from cache

  - Removing zendframework/zend-captcha (2.5.2)
  - Installing zendframework/zend-captcha (2.6.0)
    Loading from cache

  - Removing zendframework/zend-config (2.5.1)
  - Installing zendframework/zend-config (2.6.0)
    Loading from cache

  - Removing zendframework/zend-crypt (2.5.2)
  - Installing zendframework/zend-crypt (2.6.0)
    Loading from cache

  - Removing zendframework/zend-db (2.6.2)
  - Installing zendframework/zend-db (2.8.1)
    Loading from cache

  - Removing zendframework/zend-code (2.6.2)
  - Installing zendframework/zend-code (2.6.3)
    Loading from cache

  - Removing zendframework/zend-di (2.5.1)
  - Installing zendframework/zend-di (2.6.1)
    Loading from cache

  - Removing zendframework/zend-escaper (2.5.1)
  - Installing zendframework/zend-escaper (2.5.2)
    Loading from cache

  - Removing zendframework/zend-feed (2.6.0)
  - Installing zendframework/zend-feed (2.7.0)
    Loading from cache

  - Removing zendframework/zend-file (2.5.1)
  - Installing zendframework/zend-file (2.7.0)
    Loading from cache

  - Removing zendframework/zend-filter (2.5.1)
  - Installing zendframework/zend-filter (2.7.1)
    Loading from cache

  - Removing zendframework/zend-inputfilter (2.5.5)
  - Installing zendframework/zend-inputfilter (2.7.2)
    Loading from cache

  - Removing zendframework/zend-form (2.6.0)
  - Installing zendframework/zend-form (2.9.0)
    Loading from cache

  - Removing zendframework/zend-i18n (2.5.1)
  - Installing zendframework/zend-i18n (2.7.3)
    Loading from cache

  - Removing zendframework/zend-log (2.6.0)
  - Installing zendframework/zend-log (2.9.0)
    Loading from cache

  - Removing zendframework/zend-mime (2.5.1)
  - Installing zendframework/zend-mime (2.6.0)
    Loading from cache

  - Removing zendframework/zend-mail (2.5.2)
  - Installing zendframework/zend-mail (2.7.1)
    Loading from cache

  - Removing zendframework/zend-memory (2.5.1)
  - Installing zendframework/zend-memory (2.5.2)
    Loading from cache

  - Removing zendframework/zend-navigation (2.5.1)
  - Installing zendframework/zend-navigation (2.8.1)
    Loading from cache

  - Removing zendframework/zend-progressbar (2.5.1)
  - Installing zendframework/zend-progressbar (2.5.2)
    Loading from cache

  - Removing zendframework/zend-json (2.6.0)
  - Installing zendframework/zend-json (2.6.1)
    Loading from cache

  - Removing zendframework/zend-serializer (2.5.1)
  - Installing zendframework/zend-serializer (2.8.0)
    Loading from cache

  - Removing zendframework/zend-session (2.5.2)
  - Installing zendframework/zend-session (2.7.3)
    Loading from cache

  - Removing zendframework/zend-uri (2.5.1)
  - Installing zendframework/zend-uri (2.5.2)
    Loading from cache

  - Removing zendframework/zend-server (2.6.0)
  - Installing zendframework/zend-server (2.7.0)
    Loading from cache

  - Removing zendframework/zend-soap (2.5.1)
  - Installing zendframework/zend-soap (2.6.0)
    Loading from cache

  - Removing zendframework/zend-tag (2.5.1)
  - Installing zendframework/zend-tag (2.6.1)
    Loading from cache

  - Removing zendframework/zend-view (2.5.3)
  - Installing zendframework/zend-view (2.8.1)
    Loading from cache

    Skipped installation of bin bin/templatemap_generator.php for package zendframework/zend-view: name conflicts with an existing file
  - Installing psr/http-message (1.0)
    Loading from cache

  - Installing zendframework/zend-diactoros (1.3.5)
    Loading from cache

  - Removing zendframework/zend-http (2.5.3)
  - Installing zendframework/zend-http (2.5.4)
    Loading from cache

  - Installing zendframework/zend-psr7bridge (0.2.2)
    Loading from cache

  - Removing zendframework/zend-mvc (2.6.0)
  - Installing zendframework/zend-mvc (2.7.10)
    Loading from cache

  - Removing zendframework/zend-console (2.5.1)
  - Installing zendframework/zend-console (2.6.0)
    Loading from cache

  - Removing symfony/yaml (v2.8.2)
  - Installing symfony/yaml (v3.1.3)
    Loading from cache

  - Removing sebastian/exporter (1.2.1)
  - Installing sebastian/exporter (1.2.2)
    Loading from cache

  - Removing sebastian/environment (1.3.3)
  - Installing sebastian/environment (1.3.7)
    Loading from cache

  - Removing phpunit/php-timer (1.0.7)
  - Installing phpunit/php-timer (1.0.8)
    Loading from cache

  - Installing webmozart/assert (1.0.2)
    Loading from cache

  - Installing phpdocumentor/reflection-common (1.0)
    Loading from cache

  - Installing phpdocumentor/type-resolver (0.2)
    Loading from cache

  - Removing phpdocumentor/reflection-docblock (2.0.4)
  - Installing phpdocumentor/reflection-docblock (3.1.0)
    Loading from cache

  - Removing phpspec/prophecy (v1.5.0)
  - Installing phpspec/prophecy (v1.6.1)
    Loading from cache

  - Installing sebastian/resource-operations (1.0.0)
    Loading from cache

  - Installing sebastian/object-enumerator (1.0.0)
    Loading from cache

  - Removing phpunit/phpunit-mock-objects (2.3.8)
  - Installing phpunit/phpunit-mock-objects (3.2.3)
    Loading from cache

  - Installing sebastian/code-unit-reverse-lookup (1.0.0)
    Loading from cache

  - Removing phpunit/php-code-coverage (2.2.4)
  - Installing phpunit/php-code-coverage (4.0.1)
    Loading from cache

  - Installing myclabs/deep-copy (1.5.1)
    Loading from cache

  - Removing phpunit/phpunit (4.8.21)
  - Installing phpunit/phpunit (5.5.0)
    Loading from cache

  - Removing zendframework/zend-test (2.5.2)
  - Installing zendframework/zend-test (2.6.1)
    Loading from cache

  - Removing zendframework/zend-text (2.5.1)
  - Installing zendframework/zend-text (2.6.0)
    Loading from cache

  - Removing zendframework/zendxml (1.0.1)
  - Installing zendframework/zendxml (1.0.2)
    Loading from cache

  - Removing zendframework/zend-xmlrpc (2.5.1)
  - Installing zendframework/zend-xmlrpc (2.6.0)
    Loading from cache

  - Removing zendframework/zend-modulemanager (2.6.1)
  - Installing zendframework/zend-modulemanager (2.7.2)
    Loading from cache

  - Removing zfcampus/zf-api-problem (1.1.0)
  - Installing zfcampus/zf-api-problem (1.2.1)
    Loading from cache

  - Removing zfcampus/zf-content-negotiation (1.1.1)
  - Installing zfcampus/zf-content-negotiation (1.2.2)
    Loading from cache

  - Removing zfcampus/zf-configuration (1.1.0)
  - Installing zfcampus/zf-configuration (1.2.0)
    Loading from cache

  - Removing zfcampus/zf-apigility-provider (1.1.0)
  - Installing zfcampus/zf-apigility-provider (1.2.0)
    Loading from cache

  - Installing michelf/php-markdown (1.6.0)
    Loading from cache

  - Removing zfcampus/zf-apigility-documentation (1.1.0)
  - Installing zfcampus/zf-apigility-documentation (1.2.1)
    Loading from cache

  - Removing zfcampus/zf-versioning (1.1.0)
  - Installing zfcampus/zf-versioning (1.2.0)
    Loading from cache

  - Removing zfcampus/zf-rpc (1.2.0)
  - Installing zfcampus/zf-rpc (1.3.0)
    Loading from cache

  - Removing zfcampus/zf-oauth2 (1.3.1)
  - Installing zfcampus/zf-oauth2 (1.4.0)
    Loading from cache

  - Removing zendframework/zend-permissions-acl (2.5.1)
  - Installing zendframework/zend-permissions-acl (2.6.0)
    Loading from cache

  - Removing zendframework/zend-authentication (2.5.2)
  - Installing zendframework/zend-authentication (2.5.3)
    Loading from cache

  - Removing zfcampus/zf-mvc-auth (1.3.1)
  - Installing zfcampus/zf-mvc-auth (1.4.2)
    Loading from cache

  - Removing zendframework/zend-paginator (2.5.1)
  - Installing zendframework/zend-paginator (2.7.0)
    Loading from cache

  - Removing zfcampus/zf-hal (1.3.0)
  - Installing zfcampus/zf-hal (1.4.2)
    Loading from cache

  - Removing zfcampus/zf-rest (1.2.0)
  - Installing zfcampus/zf-rest (1.3.1)
    Loading from cache

  - Removing zfcampus/zf-content-validation (1.2.0)
  - Installing zfcampus/zf-content-validation (1.3.4)
    Loading from cache

  - Removing zfcampus/zf-apigility-admin-ui (1.2.2)
  - Installing zfcampus/zf-apigility-admin-ui (1.3.0)
    Loading from cache

  - Removing zfcampus/zf-apigility (1.2.1)
  - Installing zfcampus/zf-apigility (1.3.0)
    Loading from cache

  - Removing zfcampus/zf-apigility-admin (1.4.1)
  - Installing zfcampus/zf-apigility-admin (1.4.3)
    Loading from cache

  - Removing zfcampus/zf-console (1.2.0)
  - Installing zfcampus/zf-console (1.3.0)
    Loading from cache

  - Removing zfcampus/zf-deploy (1.1.0)
  - Installing zfcampus/zf-deploy (1.2.0)
    Loading from cache

  - Removing zendframework/zenddiagnostics (v1.0.8)
  - Installing zendframework/zenddiagnostics (v1.0.9)
    Loading from cache

  - Removing zendframework/zend-developer-tools (dev-master)
  - Installing zendframework/zend-developer-tools (dev-master 4d8a2e4)
    Cloning 4d8a2e4a56b9620fb986657e0e701e1e65bfc7a5 from cache

Writing lock file
Generating autoload files
```


Resolve 500 due to missing AssetManager
```
php composer.phar require rwoverdijk/assetmanager
```

4. attempt to create service [succeeds], modify service [FAILS], delete service [FAILS]

**Comment by jackdpeterson** [comment]
The hack:

Modify require-dev in composer.json
```
"zfcampus/zf-apigility-admin-ui": "1.2.2",
```


composer.phar update
```
  - Removing zfcampus/zf-apigility-admin-ui (1.3.0)
  - Installing zfcampus/zf-apigility-admin-ui (1.2.2)
    Downloading: 100%   
```

Now, re-attempt create service [succeeds], modify service [succeeds], delete service [succeeds].

**Comment by jackdpeterson** [comment]
Resolved in ^1.3.2
